### DISCUSSION TITLE
JPG compression for web. Banding around the subject/object.

### DISCUSSION THREAD

**Discussion by TNNSLAB** [ImageMagick/ImageMagick#6761] [Help] [answered]
### ImageMagick version

7.1.0-4 Q16 x86_64

### Operating system, version and so on

Mac OS 10.14.6

### Description

I am trying to compress some jpg files to keep it under 500kb. 

using this command :

/usr/local/bin/convert -strip -quality 60% -resize 1800x2400\> -density 72 -units PixelsPerInch /Users/adrian/Desktop/actions image_original.jpg image_converted.jpg

I am getting some banding on my images. See the picture attached. 

Is there a way to avoid this ?
<img width="1667" alt="banding" src="https://github.com/ImageMagick/ImageMagick/assets/46636486/dabbe533-ea4d-4684-a1f9-7abce93f375a">

![bending_for_web](https://github.com/ImageMagick/ImageMagick/assets/46636486/5593df8b-283a-471b-a4c4-e515eb978684)

**Comment by snibgo** [comment]
This compression artifact is called "banding", not "bending".

JPG has only 8 bits per channel per pixel. To avoid banding, use a format with a greater depth. But your input is also JPG, so that may not help. You might also try adding a small amount of noise.

**Reply by TNNSLAB** [reply]
Hi @snibgo sorry, fixed the artifact to banding instead of bending...thx for that...

what files would you suggest working with ? tiff ? psd ? cr2 ? 

can you please show an example of noise adding ? thank you

**Comment by snibgo** [comment, marked as answer]
The main problem is that the input is JPG. JPG is a very poor input format for image processing. The near-white gradient at top-right has only about ten levels, which become visible when you increase the contrast.

You should always start from the best images available, such as camera raw files.

An example of adding noise to reduce visible banding:
```
magick shirtNeck.jpg -attenuate 8 +noise Uniform +depth out.png
```

**Reply by TNNSLAB** [reply]
nice ! this is killing the banding...i'll play with it. interesting to learn about this ! thank you again for the useful information !
